### ISSUE TITLE
CloudFormation - All deployed stacks have StackStatus: CREATE_FAILED

### ISSUE THREAD

**Issue by Tobolov** [localstack/localstack#3375]
# Type of request: This is a ...

[X ] bug report
[ ] feature request

# Detailed description
This bug seems to only be occur when running localstack directly from the source code (not docker). I have not been able to reproduce it with the latest localstack docker image at the time of writing [8132ca0c98ca](https://hub.docker.com/layers/localstack/localstack/latest/images/sha256-8132ca0c98cae33b3c48152ae6e390838a12a07bf171ca413d115ac261ec11ee?context=explore) . 

It seems that from commit de360b516125a0280d7fcbf34c6194e0186b1d08, all stacks which I attempt to deploy fail with `Unable to resolve all CloudFormation resources after traversing dependency tree (maximum depth 40)`.

## Expected behavior
The supplied cloud formation template (see Steps to Reproduce) should deploy and describing the stack should shown `"StackStatus": "CREATE_COMPLETE"`.

## Actual behavior
When deploying the supplied cloud formation template (see Steps to Reproduce), the stack successfully creates the sns topic, however describing the stack via cli shows (note the CREATE_FAILED).
```json
{
      "StackId": "arn:aws:cloudformation:us-east-1:000000000000:stack/slack-sns-topic/11d1b133-4b3b-4b03-9b06-35272b35c03b",
      "StackName": "slack-sns-topic",
      "TemplateDescription": "None",
      "CreationTime": "2020-12-18T00:31:48+00:00",
      "StackStatus": "CREATE_FAILED"
}
```

Log from commit de360b516125a0280d7fcbf34c6194e0186b1d08 (first commit this bug is present in).
```
2020-12-18T10:16:08:INFO:localstack.services.cloudformation.cloudformation_starter: Error running stack deployment loop: 'CloudFormationBackend' object has no attribute 'set_exports'
2020-12-18T10:16:08:WARNING:bootstrap.py: Thread run method <function apply_patches.<locals>.run_dependencies_deployment_loop.<locals>.run_loop at 0x7f7791cf2280>(None) failed: Unable to resolve all CloudFormation resources after traversing dependency tree (maximum depth 40): ['SnsTopic'] Traceback (most recent call last):
  File "/home/tobolov/linrepos/localstack/localstack/utils/bootstrap.py", line 581, in run
    result = self.func(self.params)
  File "/home/tobolov/linrepos/localstack/localstack/services/cloudformation/cloudformation_starter.py", line 1210, in run_loop
    raise Exception('Unable to resolve all CloudFormation resources after traversing ' +
Exception: Unable to resolve all CloudFormation resources after traversing dependency tree (maximum depth 40): ['SnsTopic']
```

Log from commit 62091e2b4c01fd1bc09d838b84afd269d16b181e (HEAD of master branch at time of writing).
```
2020-12-18T10:25:39:INFO:localstack.services.cloudformation.cloudformation_starter: Error running stack deployment loop: 'CloudFormationBackend' object has no attribute 'set_exports'
2020-12-18T10:25:39:WARNING:bootstrap.py: Thread run method <function apply_patches.<locals>.run_dependencies_deployment_loop.<locals>.run_loop at 0x7f0c94579a60>(None) failed: Unable to resolve all CloudFormation resources after traversing dependency tree (maximum depth 40): [] Traceback (most recent call last):
  File "/home/tobolov/linrepos/localstack/localstack/utils/bootstrap.py", line 581, in run
    result = self.func(self.params)
  File "/home/tobolov/linrepos/localstack/localstack/services/cloudformation/cloudformation_starter.py", line 1216, in run_loop
    raise Exception('Unable to resolve all CloudFormation resources after traversing ' +
Exception: Unable to resolve all CloudFormation resources after traversing dependency tree (maximum depth 40): []
```
...

# Steps to reproduce
1. Deploy the following CloudFormation template:
```yaml
---
AWSTemplateFormatVersion: "2010-09-09"

Resources:
  SnsTopic:
    Type: AWS::SNS::Topic
    Properties: 
      TopicName: slack-sns-topic
```
2. Run `aws cloudformation describe-stacks` or `aws cloudformation list-stacks`

**Comment by whummer** [comment]
Thanks for reporting @Tobolov . This should be fixed in #3381 - can you please pull the latest version and give it another try? Also, when running the code locally, please make sure to frequently update the following dependency libraries:
```
pip install --upgrade localstack-client moto-ext
```
Hope that helps - please report here if the problem persists.

**Comment by Tobolov** [comment]
Hi @whummer, Thanks for getting this fixed! 
After a bit of testing it seems that this issue has mostly been fixed. I did come across one of our templates which would still fail to deploy though. I have remove as much as possible from the template while still preserving the bug.  

Snippet from log when deploying the attached template:
```
2020-12-21T13:35:21:WARNING:bootstrap.py: Thread run method <function apply_patches.<locals>.run_dependencies_deployment_loop.<locals>.run_loop at 0x7f1a4c2bdc10>(None) failed: Unable to resolve all CloudFormation resources after traversing dependency tree (maximum depth 40): [] Traceback (most recent call last):
  File "/home/tobolov/linrepos/localstack/localstack/utils/bootstrap.py", line 581, in run
    result = self.func(self.params)
  File "/home/tobolov/linrepos/localstack/localstack/services/cloudformation/cloudformation_starter.py", line 1217, in run_loop
    raise Exception('Unable to resolve all CloudFormation resources after traversing ' +
Exception: Unable to resolve all CloudFormation resources after traversing dependency tree (maximum depth 40): []
```

Template:

```yaml
---
AWSTemplateFormatVersion: "2010-09-09"

Parameters:
  RegmonSnsTopicSendEmails:
    Default: false
    Type: String
    AllowedValues: [true, false]

# <sensitive: template succeeds if removed>
Conditions: 
  ShouldSnsTopicSendEmails: !Equals [true, !Ref RegmonSnsTopicSendEmails]   
# </sensitive>

Resources:
  InstanceRole:
    Properties:
      RoleName: some-role
      ManagedPolicyArns:
        - arn:aws:iam::aws:policy/AmazonDynamoDBFullAccess
      AssumeRolePolicyDocument:
        Statement:
          - Effect: Allow
            Principal:
              Service:
                - ec2.amazonaws.com
                - ssm.amazonaws.com
            Action:
              - sts:AssumeRole
    Type: AWS::IAM::Role

  InstanceProfile:
    Type: AWS::IAM::InstanceProfile
    Properties:
      Path: "/"
      Roles:
        - Fn::ImportValue: fake

  Ec2Instance:
    Type: "AWS::EC2::Instance"
    # <sensitive: template succeeds if removed>
    DependsOn: 
      - InstanceProfile
    # </sensitive>
    Properties:
      InstanceType: 't3.small'
      ImageId: abc
      KeyName: "ec2-key-pair"
      IamInstanceProfile: !Ref InstanceProfile
      SecurityGroupIds: 
      - Fn::ImportValue: 'sg-id'
      SubnetId:
        Fn::ImportValue: 'private-sn-a'

Outputs:
  # <sensitive: template succeeds to basically anything here being removed>
  InstanceId:
    Value: !Ref Ec2Instance
    Export:
      Name: RegmonEc2InstanceId
  RoleArn:
    Value: !GetAtt InstanceRole.Arn
    Export:
      Name: RegmonRoleArn
  # </sensitive>
```

Let me know if you need any more information.

**Comment by Tobolov** [comment]
Hi @whummer, Just wondering if you saw this. Thanks!

**Comment by whummer** [comment]
Thanks for the update @Tobolov . We believe that this should be fixed in #3477 - can you please pull the `latest` image again and give it another try? Please let us know if this problem persists. Thanks!
